vmm/src/acpi.rs

```diff
@@ -139,6 +153,10 @@ impl aml::Aml for CPU {
                         &aml::ZERO
                     })],
                 ),
+                // The Linux kernel expects every CPU device to have a _MAT entry
+                // containing the LAPIC for this processor with the enabled bit set
+                // even it if is disabled in the MADT (non-boot CPU)
```

Comment:
Ok so here we expose for each CPU a _MAT buffer containing the exact same information that is exposed through the MADT, right?
Have you tried to remove the entries from the MADT? Does that break the VMM?

Just for my own understanding, is the point of the MADT to declare what is static versus the DSDT with the _MAT entries to describe what is potentially hotpluggable?

I mean in any case it cannot hurt to describe the same CPUs in each table.

Reply:
All the potential CPUs must have an MADT entry. If they are there at boot the flags value is set to 1 if it is a CPU that might be added later the value is 0. Not having the entries in the MADT table does not work.

The _MAT version duplicates the MADT version but the flags value is always 1 even for CPUs that are not yet present. I think this is not massively standardised but rather the Linux kernel convention. 

I think you could get away without having the MADT table entries for disabled CPUs (would still need them for boot CPUs) but you'd have to pass max potential CPUs onto the kernel commandline.

Reply:
I see, thanks for the clarifications.